build/parents/dependencies/pom.xml

```diff
@@ -7,26 +7,166 @@
+        <version.org.elasticsearch.compatible.not-regularly-tested.text>7.0 or 8.0</version.org.elasticsearch.compatible.not-regularly-tested.text>
+        <!-- The latest version of Elasticsearch tested against by default -->
+        <version.org.elasticsearch.latest>8.9.0</version.org.elasticsearch.latest>
+        <!-- The versions of OpenSearch advertised as compatible with Hibernate Search -->
+        <!-- Make sure to only mention tested versions here -->
+        <version.org.opensearch.compatible.regularly-tested.text>1.3 or 2.9</version.org.opensearch.compatible.regularly-tested.text>
+        <!-- These are the versions same as above, but pointing only to the major part (used in compatibility section of ES backend documentation
+          as versions that Hibernate Search is compatible with. -->
+        <!-- NOTE: Adding new major versions would require to update the compatibility table in `backend-elasticsearch-compatibility` section of `backend-elasticsearch.asciidoc`. -->
+        <version.org.opensearch.compatible.expected.text>1.x or 2.x</version.org.opensearch.compatible.expected.text>
+        <!-- The latest version of OpenSearch tested against by default -->
+        <version.org.opensearch.latest>2.9.0</version.org.opensearch.latest>
+
+        <version.com.google.code.gson>2.9.1</version.com.google.code.gson>
+        <version.software.amazon.awssdk>2.20.2</version.software.amazon.awssdk>
+        <!--
+            Jackson is used by
+                * the Elasticsearch REST client sniffer,
+                * avro
+                * and Wiremock
+            We want to align this version on whatever the REST client is using.
+            But because we also want to include it in the BOM to align versions, maven-enforcer-plugin will not catch that the
+            version is different from whatever version is in those mentioned libraries. So need to update it anytime we
+            are updating the Elasticsearch rest client version.
+        -->
+        <version.com.fasterxml.jackson>2.15.0</version.com.fasterxml.jackson>
+        <!-- slf4j: used by the AWS SDK -->
+        <version.org.slf4j>1.7.36</version.org.slf4j>
+
+        <!-- >>> ORM 6 with Jakarta Persistence -->
+        <!-- IMPORTANT: Make sure that version.jakarta.persistence aligns with the imported version! -->
+        <version.org.hibernate.orm>6.2.7.Final</version.org.hibernate.orm>
+
+        <javadoc.org.hibernate.orm.url>https://docs.jboss.org/hibernate/orm/${parsed-version.org.hibernate.orm.majorVersion}.${parsed-version.org.hibernate.orm.minorVersion}/javadocs/</javadoc.org.hibernate.orm.url>
+        <documentation.org.hibernate.orm.url>https://docs.jboss.org/hibernate/orm/${parsed-version.org.hibernate.orm.majorVersion}.${parsed-version.org.hibernate.orm.minorVersion}/userguide/html_single/Hibernate_User_Guide.html</documentation.org.hibernate.orm.url>
```

Comment:
So this property is defined in `/build/parents/dependencies/pom.xml`, but it relies on `parsed-version.org.hibernate.orm.majorVersion` which is defined in `/pom.xml`, but is generated based on `version.org.hibernate.orm` which is defined in `/build/parents/dependencies/pom.xml`.

... are you sure it's a good idea not to move the plugin config to this pom?

Reply:
Was I sure about it - no 😃, did I try that it somehow was working and was surprised to see that it does - yes 😆.
yeah, my thought was since it worked, maybe it would be easier to find the plugin configs if they stayed in the same place. But I agree that it looks strange. So I've split the executions and moved the properties related to the docs and based on lib versions to the other pom.

Reply:
You really don't want to move plugin configuration here, do you xD

I'll try something else: see the configuration of the maven-compiler-plugin in `/pom.xml`: it references the version of jboss-logging... which is defined in `/build/parents/dependencies/pom.xml`!

I really don't think it makes sense to configure plugins in a context that does not have (conceptually) access to dependency versions, since some plugins will need to refer to those versions...

If your concern is about publishing the root POM or the public BOM, then that point is moot once we'll flatten all POMs (#3603).

If your concern is about not being able to import the public BOM if we move plugin configuration here, I don't think it matters anymore, since we only expose our own artifacts in that BOM now, and you removed any chance that we will forget about one in #3605. So the BOM will always be aligned with our build.

The way I see it, considering the various ways we've had to alter our plans regarding BOMs, there are only two ways forward that make sense now:

1. Have a dedicated `build/parents/build/pom.xml` that contains all properties (versions and others), dependency management, and plugins, separated from the root POM that would basically become an "aggregator POM".
2. Move versions back to the root POM.

I can see value in 1 over 2, be it only as a way to un-clutter the root POM. Option 2 would be more in line with other projects such as Validator, but I'm not sure we even care nowadays, with some projects using Gradle and not others, and so many differences between projects. Feel free to start a thread on Zulip if you want to discuss our options.

Reply:
> You really don't want to move plugin configuration here, do you xD

was it too obvious? 😆 🤣 

Ok so I've moved most of the plugins to this new pom, some of them remained in the root, and there are comments on why they are there.
As to some moving the IDE profiles -- it somewhat makes sense to have them in the root pom, and I'd rather keep them there, but then they are referencing properties e.g. IDEA is using the jboss-logging/jmh/java-test versions ... so they got moved.
I've left the reporting in the root pom since I wasn't sure about it. and also that last profile for including reporting module ... also stayed in the root.

now a slight issue I can see is that any plugin we want to call from the command line if not defined in the root would be executed with some random version before reaching the modules that have this new one as a parent (that's why I've kept the clean/install plugins in the root).

The next logical step seems to clean up the modules 😄, meaning it might make sense to have something like a couple of modules in the root (`helpers` `boms` `actual` obviously with better names). `Helpers` would have the build configs, property collector, enforcer rules etc, basically anything that we'd later be referencing during the main build.... it would mean that some plugins like compilation or maybe for tests would have a management entry in the root pom and then configured in the modules...

Reply:
> As to some moving the IDE profiles -- it somewhat makes sense to have them in the root pom, and I'd rather keep them there, but then they are referencing properties e.g. IDEA is using the jboss-logging/jmh/java-test versions ... so they got moved.

For everything I can see in this profile (jboss-logging/jmh/java-test, dependency convergence), this makes sense, since the corresponding configuration would not be relevant for modules that do not extend the `build` parent POM  (BOM, `build/config`, `build/enforcer`, ...).

> now a slight issue I can see is that any plugin we want to call from the command line if not defined in the root would be executed with some random version before reaching the modules that have this new one as a parent (that's why I've kept the clean/install plugins in the root).

I suppose we could use plugin management to set all the plugin versions in the root POM, and in `build/parent/build/pom.xml` we'd only include plugin configuration (and only for those plugins where we need configuration)? Since only the configuration of those plugins is actually a problem to have in the root POM, not version management. I'll let you decide if this makes sense (and if you're not fed up with changing that stuff ;) ).

> Helpers would have the build configs, property collector, enforcer rules etc, basically anything that we'd later be referencing during the main build....

That's what we currently have with the `build` directory, no?

I'm not getting what needs to be cleaned up exactly.

Reply:
> > now a slight issue I can see is that any plugin we want to call from the command line if not defined in the root would be executed with some random version before reaching the modules that have this new one as a parent (that's why I've kept the clean/install plugins in the root).
> 
> I suppose we could use plugin management to set all the plugin versions in the root POM, and in `build/parent/build/pom.xml` we'd only include plugin configuration (and only for those plugins where we need configuration)? Since only the configuration of those plugins is actually a problem to have in the root POM, not version management. I'll let you decide if this makes sense (and if you're not fed up with changing that stuff ;) ).

By the way, good point about some plugins needing to be configured at the root (in the aggregator POM, essentially).

Looking closer at what you did, there seems to be a few more plugins in that case, in particular jqassistant and sonar, maybe others. Those plugins actually take advantage of the fact they're executed in the aggregator POM.

Maybe only a few plugins actually need to be in `build/parents/build/pom.xml`... essentially anything that is specific to Hibernate Search's runtime (e.g. annotation processors, ...), as opposed to the way we want modules to be built in general.

I think I'm starting to be a nuisance rather than a help here. Do you want me to have a closer look and push a commit with something that seems to make sense to me?

Reply:
See https://hibernate.zulipchat.com/#narrow/stream/132092-hibernate-search-dev/topic/HSEARCH-4861 for the rest of the discussion